packages/relay/src/routes/user.ts

```diff
@@ -0,0 +1,55 @@
+import { Express } from 'express'
+import { DB } from 'anondb/node'
+import { Synchronizer } from '@unirep/core'
+import TwitterClient from '../singletons/TwitterClient'
+import { ethers } from 'ethers';
+import crypto from 'crypto';
+
+const STATE = crypto.randomUUID()
+
+export default (app: Express, db: DB, synchronizer: Synchronizer) => {
+    app.get('/api/login', async (req, res) => {
+        res.redirect(TwitterClient.authClient.generateAuthURL({
+            state: STATE,
+            code_challenge: "BDcX6UdIZ7uAJdWnCsaKL0G9O2rwzvu4"
+        }))
+    })
+
+    app.get('/api/callback', async (req, res) => {
+        try {
+            const { code, state } = req.query;
+            if (state !== STATE)
+                return res.status(500).json({
+                    error: "State isn't matching"
+                });
+            
+            TwitterClient.authClient.requestAccessToken(code as string)
+                .then(_ => TwitterClient.client.users.findMyUser())
+                .then(async userInfo => {
+
+                    const userId = userInfo.data?.id
+                    if (!userId) throw new Error('No userId')
+                    
+                    // todo check hash function require?
+                    const hash = crypto.createHash('md5')
+                    const hashUserId = hash.update(userId).digest('hex')
```

Comment:
md5 is not a secure hash
If we don't want others find the pre-image of the hash (e.g. id), you should use secure hash function e.g. SHA3